File: src/main/java/net/gtaun/shoebill/common/player/PlayerLifecycleObject.java
package net.gtaun.shoebill.common.player;

import net.gtaun.shoebill.common.AbstractShoebillContext;
import net.gtaun.shoebill.object.Player;
import net.gtaun.util.event.EventManager;

public abstract class PlayerLifecycleObject extends AbstractShoebillContext
{
	protected final Player player;
	
	
	public PlayerLifecycleObject(EventManager eventManager, Player player)
	{
		super(eventManager);
		this.player = player;
	}
	
	public Player getPlayer()
	{
		return player;
	}
}

File: src/main/java/net/gtaun/shoebill/common/player/PlayerLifecycleObject.kt
package net.gtaun.shoebill.common.player

import net.gtaun.shoebill.Shoebill
import net.gtaun.shoebill.common.LifecycleObject
import net.gtaun.shoebill.entities.Player

/**
 * Created by marvin on 19.11.16 in project shoebill-common.
 * Copyright (c) 2016 Marvin Haschker. All rights reserved.
 */

abstract class PlayerLifecycleObject(val player: Player) : LifecycleObject(Shoebill.get().eventManager)